src/zarr/testing/strategies.py

```diff
@@ -143,3 +145,19 @@ def basic_indices(draw: st.DrawFn, *, shape: tuple[int], **kwargs):  # type: ign
             )
         )
     )
+
+
+def key_ranges(keys: SearchStrategy | None = None) -> SearchStrategy[list]:
+    """fn to generate key_ranges strategy for get_partial_values()
+    returns list strategy w/ form: [(key, (range_start, range_step)),
+                                    (key, (range_start, range_step)),...]
+    """
+    if keys is None:
+        group_st = st.text(alphabet=string.ascii_letters + string.digits, min_size=1, max_size=10)
+        keys = st.lists(group_st, min_size=1, max_size=5).map("/".join)
```

Comment:
```suggestion
def key_ranges(keys: SearchStrategy = node_names) -> SearchStrategy[list]:
    """fn to generate key_ranges strategy for get_partial_values()
    returns list strategy w/ form: [(key, (range_start, range_step)),
                                    (key, (range_start, range_step)),...]
    """
```

Let's use `numpydoc` conventions for docstrings: https://numpydoc.readthedocs.io/en/latest/